### ISSUE TITLE
Inconsistent order of plugin apps in the sidebar menu

### ISSUE THREAD

**Issue by zserge** [grafana/grafana#36310]
**What happened**:

When multiple plugins are installed -- their icons in the nav menu change order on page reloads.

**What you expected to happen**:

The order of the plugins in the menu to be consistent (sorted by name/ID? by installation timestamp?).

**How to reproduce it (as minimally and precisely as possible)**:

* Install and enable several plugins, that contain dashboards added to the navigation menu.
* Open Grafana web UI
* Refresh the page several times
* Plugin icons jump around changing their positions in the menu.

**Anything else we need to know?**:

In `pkg/plugins/manager/manager.go` there is a method PluginManager.Apps(), which iterates over a map. Iteration order is not guaranteed, so on each request the user may get different results:

https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/manager.go#L270-L272

**Environment**:
- Grafana version: main branch, built from source
- Data source type & version:
- OS Grafana is installed on: macOS Big Sur
- User OS & Browser: Chromium
- Grafana plugins: redis, worldping, "myorgid-simple-app" example.
- Others:

**Comment by zserge** [comment]
<p><img width="43" alt="Screenshot 2021-06-30 at 17 15 39" src="https://user-images.githubusercontent.com/1911760/123986762-e1348680-d9c6-11eb-8b7a-adce7a4c8792.png">
<img width="39" alt="Screenshot 2021-06-30 at 17 15 48" src="https://user-images.githubusercontent.com/1911760/123986766-e265b380-d9c6-11eb-8c68-ebf7fb4035e4.png">
</p>Here are the two screenshots on how it looks like before and after a page reload.

**Comment by marefr** [comment]
Maybe something for @wbrowne to look into/provide some feedback?

**Comment by wbrowne** [comment]
Good catch. For `/plugins` we're doing the following https://github.com/grafana/grafana/blob/main/pkg/api/plugins.go#L101 https://github.com/grafana/grafana/blob/main/pkg/api/dtos/plugins.go#L57

https://github.com/grafana/grafana/blob/main/pkg/plugins/manager/queries.go#L54 is where the apps are fetched and returned - so we could do the same there for consistency https://github.com/grafana/grafana/blob/main/pkg/plugins/models.go#L147
